internal/service/invoicing/invoice_unit.go

```diff
@@ -0,0 +1,257 @@
+	LastModified           timetypes.RFC3339                          `tfsdk:"last_modified"`
+	Name                   types.String                               `tfsdk:"name"`
+	Rule                   fwtypes.ListNestedObjectValueOf[ruleModel] `tfsdk:"rule"`
+	TaxInheritanceDisabled types.Bool                                 `tfsdk:"tax_inheritance_disabled"`
+	Tags                   tftags.Map                                 `tfsdk:"tags"`
+	TagsAll                tftags.Map                                 `tfsdk:"tags_all"`
+}
+
+type ruleModel struct {
+	LinkedAccounts fwtypes.SetValueOf[types.String] `tfsdk:"linked_accounts"`
+}
+
+func (r *invoiceUnitResource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	var data invoiceUnitResourceModel
+	response.Diagnostics.Append(request.Plan.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	conn := r.Meta().InvoicingClient(ctx)
+
+	input := invoicing.CreateInvoiceUnitInput{}
+	response.Diagnostics.Append(fwflex.Expand(ctx, data, &input)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	input.ResourceTags = getTagsIn(ctx)
+
+	output, err := conn.CreateInvoiceUnit(ctx, &input)
+	if err != nil {
+		response.Diagnostics.AddError("creating Invoice Unit", err.Error())
+		return
+	}
+
+	data.ID = fwflex.StringToFramework(ctx, output.InvoiceUnitArn)
+	data.ARN = fwflex.StringToFramework(ctx, output.InvoiceUnitArn)
+
+	// Read the resource to get computed values
+	invoiceUnit, err := findInvoiceUnitByARN(ctx, conn, data.ID.ValueString())
+	if err != nil {
+		response.Diagnostics.AddError("reading Invoice Unit after create", err.Error())
+		return
+	}
+
+	response.Diagnostics.Append(fwflex.Flatten(ctx, invoiceUnit, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	data.ID = fwflex.StringToFramework(ctx, invoiceUnit.InvoiceUnitArn)
+	data.ARN = fwflex.StringToFramework(ctx, invoiceUnit.InvoiceUnitArn)
+
+	response.Diagnostics.Append(response.State.Set(ctx, data)...)
+}
+
+func (r *invoiceUnitResource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	var data invoiceUnitResourceModel
+	response.Diagnostics.Append(request.State.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	conn := r.Meta().InvoicingClient(ctx)
+
+	output, err := findInvoiceUnitByARN(ctx, conn, data.ID.ValueString())
+	if tfresource.NotFound(err) {
+		response.Diagnostics.Append(fwdiag.NewResourceNotFoundWarningDiagnostic(err))
+		response.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		response.Diagnostics.AddError("reading Invoice Unit", err.Error())
+		return
+	}
+
+	response.Diagnostics.Append(fwflex.Flatten(ctx, output, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	data.ID = fwflex.StringToFramework(ctx, output.InvoiceUnitArn)
+	data.ARN = fwflex.StringToFramework(ctx, output.InvoiceUnitArn)
```

Comment:
Can the `WithFieldNamePrefix` option be used in the `Flatten` call above to get these to match without these extra steps?

https://github.com/hashicorp/terraform-provider-aws/blob/319faeb1b9eb0cf369c0c967e3a2b200fd375c77/internal/framework/flex/options.go#L32-L41